hs_core/views/resource_folder_hierarchy.py

```diff
@@ -152,11 +162,23 @@ def data_store_structure(request):
             if 'url' in f.logical_file.extra_data:
                 f_ref_url = f.logical_file.extra_data['url']
 
+            # check if this file (f) is part of a model program folder aggregation
+            if logical_file_type == "ModelProgramLogicalFile":
+                if f.file_folder is not None and f.logical_file.folder is not None:
+                    if f.file_folder.startswith(f.logical_file.folder):
+                        has_model_program_aggr_folder = True
+            elif logical_file_type == "ModelInstanceLogicalFile":
+                if f.file_folder is not None and f.logical_file.folder is not None:
+                    if f.file_folder.startswith(f.logical_file.folder):
+                        has_model_instance_aggr_folder = True
```

Comment:
I don't see what this new block buys us.  logical_file_type should be enough to pass to the frontend.  If we have a model logical file that doesn't meet these criteria... Is that model logical file in a bad state?